gamla/graph.py

```diff
@@ -8,29 +8,63 @@
 from gamla import functional, functional_generic
 
 
+@toolz.curry
+def _is_better_node(node: Any, seen: Dict, key: Callable):
+    old_node = seen[key(node)]
+    return node > old_node
+
+
+@toolz.curry
+def _save_to_dict(element: Any, dictionary: Dict, func: Callable):
+    dictionary[func(element)] = element
+
+
+@toolz.curry
+def _accumulate(
```

Comment:
not a good name, this function returns a boolean, so maybe something with `is_` prefix